File: backend/Kfoodle/Kfoodle.Contracts/Requests/Test/GetEditQuestions/Question.cs
using Kfoodle.Contracts.Enums;

namespace Kfoodle.Contracts.Requests.Test.GetEditQuestions;

/// <summary>
/// Вопрос
/// </summary>
public class Question
{
    /// <summary>
    /// Id
    /// </summary>
    public int Id { get; set; }

    /// <summary>
    /// Текст вопроса
    /// </summary>
    public string QuestionText { get; set; } = null!;
    
    /// <summary>
    /// Тип вопроса
    /// </summary>
    public QuestionType QuestionType { get; set; }
    
    /// <summary>
    /// Правильный ответ
    /// </summary>
    public string? RightAnswer { get; set; }

    /// <summary>
    /// Варианты
    /// </summary>
    public List<Choice>? Choices { get; set; } = new();
}

File: frontend/kfoodle/src/hooks/types/Question.ts
export enum QuestionType {
  SingleAnswer = 0,
  MultipleAnswers,
  InputAnswer,
  SequenceAnswer,
}
``;

export type Choice = {
  id: number;
  choiceText: string;
  isCorrect: boolean;
};

export type Question = {
  id: number;
  questionText: string;
  questionType: QuestionType;
  choices: Choice[];
  rightAnswer: string;
};

export type SingleAnswerQuestion = Question & {
  choices: Choice[];
};

export type MultipleAnswersQuestion = Question & {
  choices: Choice[];
};

export type InputAnswersQuestion = Question & {
  rightAnswer: string;
};
